File: src/main/java/org/tindalos/principle/infrastructure/service/jdepend/MetricsCalculator.java
package org.tindalos.principle.infrastructure.service.jdepend;

import jdepend.framework.JavaPackage;

import org.tindalos.principle.domain.core.Metrics;

public class MetricsCalculator {
    
    public Metrics calculate(JavaPackage jPackage) {
        return new Metrics(jPackage.afferentCoupling(), jPackage.efferentCoupling(), jPackage.abstractness(), jPackage.instability(), jPackage.distance());
    }

}

File: src/main/java/org/tindalos/principle/infrastructure/service/jdepend/MetricsCalculator.scala
package org.tindalos.principle.infrastructure.service.jdepend

import org.tindalos.principle.domain.core.Metrics
import jdepend.framework.JavaPackage

class MetricsCalculator {

  def calculate(jPackage: JavaPackage) =
    new Metrics(jPackage.afferentCoupling(), jPackage.efferentCoupling(), jPackage.abstractness(), jPackage.instability(), jPackage.distance())
}